app/src/main/java/com/hedvig/app/ui/compose/composables/BlurredFullScreenProgressOverlay.kt

```diff
@@ -0,0 +1,155 @@
+    AnimatedCircles()
+    content()
+  }
+}
+
+@Composable
+private fun AnimatedCircles() {
+  val transition = rememberInfiniteTransition()
+  val configuration = LocalConfiguration.current
+
+  val translationX by transition.animateValue(
+    initialValue = Random.nextInt(-200..-100).dp,
+    targetValue = Random.nextInt(40..300).dp,
+    typeConverter = Dp.VectorConverter,
+    animationSpec = infiniteRepeatable(
+      animation = tween(durationMillis = 12000, easing = FastOutSlowInEasing),
+      repeatMode = RepeatMode.Reverse,
+    ),
+  )
+
+  val translationY by transition.animateValue(
+    initialValue = Random.nextInt(50..500).dp,
+    targetValue = Random.nextInt(-500..-400).dp,
+    typeConverter = Dp.VectorConverter,
+    animationSpec = infiniteRepeatable(
+      animation = tween(durationMillis = 10000, easing = FastOutSlowInEasing),
+      repeatMode = RepeatMode.Reverse,
+    ),
+  )
+
+  val translationX2 by transition.animateValue(
+    initialValue = Random.nextInt(-200..0).dp,
+    targetValue = Random.nextInt(0..100).dp,
+    typeConverter = Dp.VectorConverter,
+    animationSpec = infiniteRepeatable(
+      animation = tween(durationMillis = 8000, easing = FastOutSlowInEasing),
+      repeatMode = RepeatMode.Reverse,
+    ),
+  )
+
+  val translationY2 by transition.animateValue(
+    initialValue = Random.nextInt(-200..-50).dp,
+    targetValue = Random.nextInt(50..200).dp,
+    typeConverter = Dp.VectorConverter,
+    animationSpec = infiniteRepeatable(
+      animation = tween(durationMillis = 15000, easing = FastOutSlowInEasing),
+      repeatMode = RepeatMode.Reverse,
+    ),
+  )
+
+  val radius by transition.animateValue(
+    initialValue = Random.nextInt(150..200).dp,
+    targetValue = Random.nextInt(220..250).dp,
+    typeConverter = Dp.VectorConverter,
+    animationSpec = infiniteRepeatable(
+      animation = tween(durationMillis = 10000, easing = FastOutSlowInEasing),
+      repeatMode = RepeatMode.Reverse,
+    ),
+  )
+
+  val radius2 by transition.animateValue(
+    initialValue = Random.nextInt(70..80).dp,
+    targetValue = Random.nextInt(150..300).dp,
+    typeConverter = Dp.VectorConverter,
+    animationSpec = infiniteRepeatable(
+      animation = tween(durationMillis = 16000, easing = FastOutSlowInEasing),
+      repeatMode = RepeatMode.Reverse,
+    ),
+  )
+
+  Canvas(
+    modifier = Modifier
+      .fillMaxSize()
+      .blur(80.dp, BlurredEdgeTreatment.Rectangle),
+  ) {
+
+    val screenHeight = configuration.screenHeightDp.dp.toPx()
+    val screenWidth = configuration.screenWidthDp.dp.toPx()
+
+    translate(
+      left = min(translationX.toPx(), screenWidth),
+      top = min(translationY.toPx(), screenHeight),
+    ) {
+      drawCircle(progressBlue, radius = radius.toPx())
+    }
+
+    translate(
+      left = min(translationX2.toPx(), screenWidth),
+      top = min(translationY2.toPx(), screenHeight),
+    ) {
+      drawCircle(progressYellow, radius = radius2.toPx())
+    }
+  }
```

Comment:
Instead of having to do a min() here to make sure it's within bounds, maybe we could make sure that the translation values don't go outside in the first place. That would also help in the case that let's say half of the value is outside of the screen, and the other half is inside, for the entire process when the value is outside of the screen's bounds it would look like it's not animating since it'd be forced to be inside the bounds.
Also the min only works if we're going outside the bounds on the right side, what if the value would go to the minus, then the min wouldn't do anything right?

I may be missing smth with the numbers here of course, but these are just some thoughts from looking at the code.

Reply:
Could not really find a better alternative to doing it like this. Let's revisit if we find the animation looks off on other devices.